syntax = "proto3";


service SoaMafiaServer {
    rpc RegistratePlayer(Player) returns (PlayersList) {}
    rpc NotificationStream(Player) returns (stream Notifications){}
    rpc Disconnect(Player) returns (Empty){}
    rpc StartGame(Player) returns (StartGameResponse){}
    rpc Ready(Player) returns (Empty){}
    rpc SendVote(Vote) returns (VoteResponse){}
    rpc MafiaChoice(Vote) returns (VoteResponse){}
    rpc CommissionerChoice(Vote) returns (VoteResponse){}
    rpc AlivePlayers(Room) returns (PlayersList){}
    rpc MafiaPlayers(Room) returns (PlayersList){}
}

message Room {
    string room_name = 1;
}

message Player {
    string name = 1;
    string room_name = 2;
}

message PlayersList {
    repeated string players = 1;
}

message Notifications {
    string name = 1;
    string type = 2;
}

message StartGameResponse {
    repeated string players = 1;
    string role = 2;
}

message Vote {
    string name = 1;
    string room_name = 2;
    string vote = 3;
}

message VoteResponse {
    string player = 1;
    string role = 2;
}


message Empty{

}